### ISSUE TITLE
Missing examples of using with Tornado

### ISSUE THREAD

**Issue by jonathansp** [argaen/aiocache#128]
Does aiocache work with tornado? What am I missing here?

```python
import asyncio
import tornado.web
import tornado.ioloop
from aiocache import cached
from datetime import datetime

class MainHandler(tornado.web.RequestHandler):

    @cached(ttl=10)
    def time(self):
        return datetime.now().isoformat()

    async def get(self):
        self.write(await self.time())

if __name__ == "__main__":
    tornado.ioloop.IOLoop.configure('tornado.platform.asyncio.AsyncIOLoop')
    app = tornado.web.Application([(r"/", MainHandler)])
    app.listen(7777)
    tornado.ioloop.IOLoop.current().start()
```

**Comment by argaen** [comment]
Can you post the tornado version you are using?

Also, are receiving an error or its just not working?

**Comment by jonathansp** [comment]
tornado (4.4.2).

It seems aiocache stay stuck waiting for await self.time(). Do I need to pass tornado's ioloop to aiocache?

**Comment by argaen** [comment]
So, current status. I can reproduce in python 3.5.2 but works fine in python 3.6. I've added a couple of things to make it work:

```python
import tornado.web
import tornado.ioloop
from aiocache import cached
from datetime import datetime


class MainHandler(tornado.web.RequestHandler):

    @cached(ttl=10, key="MainHandlerTime")
    async def time(self):
        return datetime.now().isoformat()

    async def get(self):
        self.write(await self.time())


if __name__ == "__main__":
    tornado.ioloop.IOLoop.configure('tornado.platform.asyncio.AsyncIOLoop')
    app = tornado.web.Application([(r"/", MainHandler)])
    app.listen(7777)
    tornado.ioloop.IOLoop.current().start()
```

I'll keep investigating why doesn't work on python 3.5.2. Also I've opened a new issue #129 (that's why I added the `key` keyword in your time function).

**Comment by argaen** [comment]
Apparently it's a problem between asyncio timeouts and tornado which looks its solved in python 3.6. https://github.com/python/asyncio/issues/392

I'll try to find a workaround and make it work in both python versions. If not possible there is always thehttps://github.com/python/asyncio/issues/392 possibility of not calling the `wait_for` if the class timeout is 0. I've done a quick patch release in order to fix this. To make it work just call the decorator as `@cached(ttl=10, key="MainHandlerTime", timeout=0)`. In case you need to use the class explicitly pass the `timeout` arg too.

**Comment by argaen** [comment]
I've opened an issue in Tornado https://github.com/tornadoweb/tornado/issues/1953. Closing since this is more asyncio - tornado related. Once I get an answer I'll update the example if necessary.
